actor/root_context.go

```diff
@@ -0,0 +1,116 @@
+package actor
+
+import "time"
+
+type RootContext struct {
+	senderMiddleware SenderFunc
+	headers          messageHeader
+}
+
+var emptyRootContext = &RootContext{
+	senderMiddleware: nil,
+	headers:          emptyMessageHeader,
+}
+
+func EmptyRootContext() *RootContext {
```

Comment:
Would be great if the `RootContext` wasn't something too special, like right now. I got several places where I want either a `Context` or a `RootContext` because I want to be able to Send or Spawn.
I played with adding an `OuterContext` like this

```
type OuterContext interface {
	SpawnerContext
	SenderContext
}
```

which I use instead of `RootContext` and that solves most of my problems, since that allows for using either a normal Context or a RootContext, while still being able to both Spawn and Send.

Reply:
Could be that this `OuterContext` could be called `InitiatingContext`, since it's always about getting stuff started

Reply:
@emilingerslev Hmm, strange...Did I implement something wrong? I just double checked the RootContext and it indeed implmenets context for spawner and sender.

Reply:
I would also opt to tighten the root context. The root context only needs to be able to send and spawn.
Therefore it should only have the `sender` and `spawner` interface.

I would also opt for moving the message and messageHeader into a separated message interface. That would remove the ability to access a message from the root context.